### PR TITLE
[Bugfix] Modify the logic of obtaining image ref when deleting snapshots

### PR DESCRIPTION
**Pull request by power-more**
In some cases, such as a user-defined label in the upper layer, the snapshot label is not empty, but does not contain
"containerd.io/snapshot/cri.image-ref", the function should continue to execute normally instead of returning an error.

Signed-off-by: zhaoshang <zhaoshangsjtu@linux.alibaba.com>

### CODE REVIEW

File: snapshot/snapshot.go [hunk 1]
```diff
@@ -403,12 +403,14 @@ func (o *snapshotter) Remove(ctx context.Context, key string) error {
 	}
 
 	if snap.Labels != nil {
-		if imageID, ok := snap.Labels[label.CRIImageRef]; ok {
-			if err := o.fs.DelSnapshot(imageID); err != nil {
-				return errors.Wrap(err, "failed to delete snapshot in cache manager")
+		if _, ok := snap.Labels[label.TargetSnapshotRef]; ok {
+			if imageID, ok := snap.Labels[label.CRIImageRef]; ok {
```
Comment (changweige) [inline snapshot/snapshot.go:407]: Why need to check if a particular label existed for the deleted snapshot?  What's the problem if we delete a snapshot which does not meet the labels combination?

Comment (imeoer) [reply snapshot/snapshot.go:407]: Does it work for OCI image layer?

**Comment by changweige** [comment]
LGTM

**Review by changweige — APPROVED** [review]

File: snapshot/snapshot.go (hunk 1 above)
Comment (changweige) [reply snapshot/snapshot.go:409]: As discussed with @power-more , it does not affect the OCI layer.

**Pull request merged**